protocol/protocol.go

```diff
@@ -208,35 +208,44 @@ func (c *Chain) AddBlockCallback(f BlockCallback) {
 	c.blockCallbacks = append(c.blockCallbacks, f)
 }
 
-// WaitForBlockSoon waits for the block at the given height,
+// WaitForBlockSoon returns a channel that
+// waits for the block at the given height,
 // but it is an error to wait for a block far in the future.
 // WaitForBlockSoon will timeout if the context times out.
 // To wait unconditionally, the caller should use WaitForBlock.
-func (c *Chain) WaitForBlockSoon(ctx context.Context, height uint64) error {
-	const slop = 3
-	if height > c.Height()+slop {
-		return ErrTheDistantFuture
-	}
+func (c *Chain) WaitForBlockSoon(ctx context.Context, height uint64) chan error {
+	ch := make(chan error)
```

Comment:
This needs to be buffered so we don't leak a goroutine in the case the caller selects a different channel.